contrib/repairStatus.py

```diff
@@ -0,0 +1,199 @@
+
+  log.info("Total ranges: curl %(url)s -d '%(query)s'" % dict(url=url, query=query))
+  r = requests.get(url, data=query )
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    message = json.dumps(response['hits']['hits'][0]['fields']['message'])    
+    total_ranges = int(re.search('(?<=repairing\s)\d+(?=\sranges)', message).group(0))
+    return total_ranges
+
+
+def print_totals(node, days, total_successful_range_repairs, total_to_repair):
+  percent_complete = total_successful_range_repairs * 100 / total_to_repair
+  print ("Checking repair messages for", node, "for the past", days, "days")
+  print ("Count is: " + str(total_successful_range_repairs))
+  print ("Total to repair is: " + str(total_to_repair))
+  print ("Percent Complete: " + str(percent_complete) + "%")
+
+
+def main():
+  parser = argparse.ArgumentParser(
+    description = '''This script shows the current status of a repair running on a Cassandra node.  
+                    Elasticsearch is required so that we can query efficiently.
+
+                    Note that NODE is required, and it should be any part of the hostname
+                    of the node that will uniquely identify the node.''')
+  parser.add_argument('-n', '--node',     type=str,
+                                          required=True, 
+                                          help='''Which node to check. This can be any portion 
+                                                  of the hostname that uniquely identifies the node. 
+                                                  (required)''')
+  parser.add_argument('--host',           type=str, 
+                                          help='Hostname for Elasticsearch (required)', 
+                                          default='localhost')
+  parser.add_argument('-p', '--port',     type=int, 
+                                          help='''Port on host where Elasticsearch is running 
+                                                  (default: 9200)''', 
+                                          default=9200)
+  parser.add_argument('-d', '--days',     type=int, 
+                                          help='''How many days of logs to search. If the script fails to return 
+                                                  data, try increasing the default value here 
+                                                  (default: 3)''', 
+                                          default=3)
+  parser.add_argument('-v', '--verbose',  action='store_true',
+                                          help='Display verbose information')
+  
+  args = parser.parse_args()
+
+  if args.verbose:
+      log.basicConfig(format="%(levelname)s: %(message)s", level=log.DEBUG)
+  else:
+      log.basicConfig(format="%(levelname)s: %(message)s")
+  
+  # Set the current time and the point in time to start searching from
+  current_time=datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.000Z')
```

Comment:
It's not really important for this particular code since it's dealing with such big ranges, but one thing to keep in mind when generating multiple subsequent timestamps is that you should generally call `utcnow` only once and store it in a temporary.

I also did a little digging and found that you *don't* in fact need to copy and paste that hideous format string all over the place; `datetime` has a handy `isoformat()` method.

```
>>> datetime.datetime.utcnow().isoformat()
'2015-01-09T08:32:07.112879'
```